secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/jws/signer/CompactSerializationJwsSigner.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.jws.signer;
+
+import static org.forgerock.openig.secrets.SecretsProviderHeaplet.secretsProvider;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+import org.forgerock.json.jose.builders.JwsHeaderBuilder;
+import org.forgerock.json.jose.builders.JwtBuilderFactory;
+import org.forgerock.json.jose.jws.JwsAlgorithm;
+import org.forgerock.json.jose.jws.SigningManager;
+import org.forgerock.json.jose.jws.handlers.SigningHandler;
+import org.forgerock.json.jose.jwt.JwtClaimsSet;
+import org.forgerock.openig.heap.GenericHeaplet;
+import org.forgerock.openig.heap.HeapException;
+import org.forgerock.secrets.Purpose;
+import org.forgerock.secrets.SecretsProvider;
+import org.forgerock.secrets.keys.SigningKey;
+import org.forgerock.util.promise.Promise;
+import org.forgerock.util.promise.Promises;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link JwsSigner} to sign a {@link Map} message.s<br/>
+ * Provides a Heaplet functionality to be added in the Heap as object for use by the filters and/or the IG scripts<br/>
+ * @see CompactSerializationJwsSigner.Heaplet
+ */
+public class CompactSerializationJwsSigner implements JwsSigner {
+
+    private static final Logger logger = LoggerFactory.getLogger(CompactSerializationJwsSigner.class);
+    private final SigningManager signingManager;
+    private final Purpose<SigningKey> signingKeyPurpose;
+    private final String algorithm;
+    private final String kid;
+
+    public CompactSerializationJwsSigner(
+            SecretsProvider secretsProvider,
+            String signingKeyId,
+            String kid,
+            String algorithm
+    ) {
+        this.signingManager = new SigningManager(secretsProvider);
+        this.kid = kid;
+        this.algorithm = algorithm;
+        this.signingKeyPurpose = Purpose.purpose(signingKeyId, SigningKey.class);
+    }
+
+    /**
+     * Compute the payload signature
+     * <p>
+     *   <ul>
+     *       <li>Accepts a JSON-Based structure payload as a {@link Map} structure</li>
+     *       <p>
+     *          Payload must not be null or empty.
```

Comment:
This info should be in the @param payload section

Reply:
done